src/plugins/events/events.ts

```diff
@@ -34,31 +35,39 @@ export class Events {
   flow: Game['flow'];
   playerID: PlayerID | undefined;
   dispatch: Array<{
-    key: string;
+    type: string;
     args: any[];
     phase: string;
     turn: number;
   }>;
+  maxEndedTurnsPerAction: number;
+  initialTurn: number;
+  currentPhase: string;
+  currentTurn: number;
 
-  constructor(flow: Game['flow'], playerID?: PlayerID) {
+  constructor(flow: Game['flow'], ctx: Ctx, playerID?: PlayerID) {
     this.flow = flow;
     this.playerID = playerID;
     this.dispatch = [];
+    this.initialTurn = ctx.turn;
+    this.updateTurnContext(ctx);
+    // This is an arbitrarily large upper threshold, which could be made
+    // configurable via a game option if the need arises.
+    this.maxEndedTurnsPerAction = ctx.numPlayers * 10;
```

Comment:
For 2 players, that's 20, which seems a high number, but not too high. Maybe 200 would be better?

Reply:
Sure. I was thinking about setting it low and seeing if people bump up against it, but as hitting this should really indicate an infinite loop error I think having it higher is reasonable — hopefully games won’t be going into production churning through hundreds of turn endings for fun.